lib/synapse.rb

```diff
@@ -85,15 +76,30 @@ def reconfigure!
       @config_updated = true
     end
 
+    def available_generators
+      Hash[@config_generators.collect{|cg| [cg.name, cg]}]
```

Comment:
eh, i feel like it's pretty standard practice to memoize in ruby if you need a specific view of otherwise-existing state. but UP2U!